Document registration update behavior

anmazzotti opened the pull request:
Resolves rancher/elemental-operator#471

kkaempf commented on docs/hardwarelabels.md:
```diff
@@ -6,9 +6,12 @@ title: ''
-This data can be used for easy identification and selection via a [MachineSelector](machineinventoryselectortemplate-reference.md).
+This data can be used for easy identification and selection via a [MachineSelector](machineinventoryselectortemplate-reference.md).  
+
+Note that Elemental Teal nodes will run `elemental-register` at each boot and every 24 hours after that, so that labels will eventually be updated to new values.  
```
That effectively means that (label) changes in `MachineRegistration` are re-synced to `MachineInventory` ?!

anmazzotti replied:
This is already the case actually, it should be documented on the MachineRegistration side.
https://elemental.docs.rancher.com/next/machineregistration-reference#machineinventorylabels

kkaempf commented:
This PR explains the _what_ but not the _why_
Imho the latter is the more important part. We should at least mention the reasoning from rancher/elemental-operator#434

anmazzotti commented:
@kkaempf  I revamped the Inventory Management section to describe a little bit what, how and why.

nunix commented on docs/customizing.md:
```diff
@@ -44,11 +44,8 @@ To apply this pattern, the ISO needs to include:
-
-The `elemental-register` command will also load the Elemental client configuration file
-located in the `/elemental` directory and/or multiple yaml files inside the `/elemental/config.d` folder.
+[Elemental client](https://github.com/rancher/elemental-toolkit/blob/main/docs/elemental.md) `install`, `upgrade` and `reset` commands can be configured with a custom [configuration file](https://rancher.github.io/elemental-toolkit/docs/customizing/general_configuration/) located in the `/elemental/config.yaml`.  
+Multiple yaml files inside the `/elemental/config.d` folder will be loaded too.  
```
```suggestion
[Elemental client](https://github.com/rancher/elemental-toolkit/blob/main/docs/elemental.md) `install`, `upgrade` and `reset` commands can be configured with a [custom configuration file](https://rancher.github.io/elemental-toolkit/docs/customizing/general_configuration/) located by default in `/elemental/config.yaml` or, if you have multiple yaml files, the `/elemental/config.d` directory will be loaded too.  
```

nunix commented on docs/hardwarelabels.md:
```diff
@@ -6,9 +6,10 @@ title: ''
 When a node is registered, hardware data is collected and made available to the MachineRegistration in the same way as [SMBIOS data](smbios.md).
 
-This data can be used for easy identification and selection via a [MachineSelector](machineinventoryselectortemplate-reference.md).
+This data can be used for easy identification and selection via a [MachineSelector](machineinventoryselectortemplate-reference.md).  
```
```suggestion
This data can be used for easy identification and selection via a [MachineSelector](machineinventoryselectortemplate-reference.md). 
```

nunix commented on docs/inventory-management.md:
```diff
@@ -11,55 +11,92 @@ the mapping of the machine to it's configuration and assigned cluster.
 
+`MachineRegistration` holds information on how to install and configure all connected Elemental Teal machines.  
+
+It is possible to update the `spec.machineInventoryLabels` and `spec.machineInventoryAnnotations`, so that they will be applied to all registered machines.  
+By default, Elemental Teal machines will attempt a registration update every 24 hours to update labels and annotations.  
```
```suggestion
It's possible to update the `spec.machineInventoryLabels` and `spec.machineInventoryAnnotations` and this will be applied to all registered machines.
By default, Elemental Teal machines will attempt a registration update every 24 hours to update labels and annotations.
```

nunix commented on docs/inventory-management.md:
```diff
@@ -11,55 +11,92 @@ the mapping of the machine to it's configuration and assigned cluster.
+It is possible to update the `spec.machineInventoryLabels` and `spec.machineInventoryAnnotations`, so that they will be applied to all registered machines.  
+By default, Elemental Teal machines will attempt a registration update every 24 hours to update labels and annotations.  
+
+While it is possible to modify the `spec.config` definition, updates to the `spec.config` will be ignored by machines that already completed installation.  
+Machines that could not complete installation will try again every 30 minutes by default, reloading the remote `MachineRegistration` definition. This can be useful to correct `spec.config` mistakes that prevent successful installation (for ex. `spec.config.elemental.install.device`), without having to create a new `MachineRegistration` and a new ISO.  
```
```suggestion
While it's possible to modify the `spec.config` definition, updates to the `spec.config` will be ignored by machines that already completed installation.
Machines that couldn't complete the installation will try again every 30 minutes by reloading the remote `MachineRegistration` definition. This can be useful to correct `spec.config` mistakes that prevent successful installation (for ex. `spec.config.elemental.install.device`), without having to create a new `MachineRegistration` and a new ISO.  
```

nunix replied:
An admonition (info) could be used here to describe the reloading time as default, and also inform where and how to change it.

nunix reviewed (commented):
Changed some phrases to active voice.
Corrected some Markdown linting.
Added a suggestion to add an admonition (info).

anmazzotti commented:
> Changed some phrases to active voice. Corrected some Markdown linting. Added a suggestion to add an admonition (info).

Thank you very much.
Regarding the info block, this timer is not (yet?) configurable.

davidcassany reviewed (approved):
Looks good

anmazzotti commented on docs/inventory-management.md:
```diff
@@ -11,55 +11,92 @@ the mapping of the machine to it's configuration and assigned cluster.
+Upon successful registration, the `MachineInventory` will inherit all `machineInventoryLabels` defined in the related `MachineRegistration`.  
+Additionally, the machine `annotations` will also be updated on each successful registration.  
+
+By default, Elemental Teal machines will attempt a registration update every 24 hours to update labels and annotations.  
```
It does attempt registration at boot, but most likely will be skipped due to the 24 hours timer on the last successful updated. 
I omitted it to not mislead users who will otherwise expect an update every boot.
I think we can address it here: https://github.com/rancher/elemental-operator/issues/481
